### ISSUE TITLE
Nvda problem with Thunderbird 60 viewing attachment in compose window

### ISSUE THREAD

**Issue by lstar4ever** [nvaccess/nvda#8698]
### Steps to reproduce:
In Thunderbird 60 after ctrl+n to compose new email and attached several files to the email,
pressing tab several time, after the subject field will arrive at the attachment field,
now NVDA doesn't read the information (file name, file size, etc) of the attachments.
pressing up and down arrow has no responce either
  
### Actual behavior:
didn't read file name and file size 
### Expected behavior:
read file name and file size etc.
press up and down arrow will read different attachment information
### NVDA Installed/portable/running from source:
latest stable 2018.2.1 installed version

#### NVDA version:
2018.2.1 
#### Name and version of other software in use when reproducing the issue:
thunderbird 60

#### Does the issue still occur after restarting your PC?
yes
#### Have you tried any other versions of NVDA?
no

**Comment by DrSooom** [comment]
CC: @MarcoZehe, @jcsteh

@mltony: As you already have a prepared Thunderbird Portable 60.0 with an existing e-mail account, could you please confirm this issue – instead of me. Thanks.

**Comment by mltony** [comment]
@DrSooom, I confirm this bug in Thunderbird 60.0 portable. Adding one attachment is enough to reproduce the issue. On older Thunderbird (52.9 in my case) it works as expected.
That control is a list view and it looks like NVDA cannot get the contents of this list view for some reason.
I'm using NVDA 2018.2.

**Comment by devil418** [comment]
This issue is much more severe. My Thunderbrid has just been updated to version 60 and it doesn't even read the attachments list in the normal message view (AKA when opening a message someone else has sent us). This makes Thunderbird almost unusable for me, as I deal with attachments a lot. I believe this should get highest possible priority, as receiving attachments is a very important function of Thunderbird, and Thunderbrid is the recommended email client for NVDA.

**Comment by devil418** [comment]
For me, a dirty workaround was to navigate to the attachment with object nav and then perform a fast doubleclick with the mouse (AKA route mouse to navigator and then double click). This is still only a workaround, though.

**Comment by jcsteh** [comment]
I wouldn't necessarily say any particular email client is officially recommended, at least not any more. This might have been the case years ago, but since then, there are other options that NVDA supports quite well (Outlook, Windows 10 Mail, various web mail apps, etc.). Thunderbird is no longer managed or developed by Mozilla Corporation, which means there is no official accessibility oversight now.

If I recall correctly (I haven't used Thunderbird in years), there used to be an attachments sub-menu in the File menu on the menu bar which allowed you to access attachments in a message you are reading. Is that not still the case? If so, that would be an alternative to the inaccessible list when reading messages (but not when composing).

**Comment by DrSooom** [comment]
@devil418: When you move to the end of the message, in browse mode press TAB once and you will reach an unlabelled button – before the attached file button begins. Press Space here to toggle it to the state "pressed, not selected" instead of "not selected". Then a list will appear. If you mean that NVDA isn't able to read the entries in this list here in Thunderbird 60.3.1 in compare to 52.9.1, I'm able to confirm this with a subscripted Podcast-RSS-Feed too.

Otherwise you will find the submenu entry "Attachments" in the Message menu – or however this menu entry is called in English. Note that the submenu list here will only be updated after opening a message.

If desired I can set up a new, clean portable copy of Thunderbird 60.3.1 in English with my test e-mail account. But currently we are talking about to different issues in Thunderbird 60.3.1 here at the same time. I guess that's not really good.

CC: @MarcoZehe

**Comment by MarcoZehe** [comment]
Sorry, am afraid I cannot help here. Thunderbird is no longer maintained by Mozilla itself, it's a community project under a different umbrella now, and I am not working for that organization. I also no longer use Thunderbird myself, stopped using it a short while ago.

**Comment by bdorer** [comment]
NVDA fokuses only the attachment list, not the entries. The entries are available with object navigation so this shouldn't be hard to work arround in the thunderbird appmodule.

**Comment by Adriani90** [comment]
cc: @novalis7747

**Comment by jcsteh** [comment]
> NVDA fokuses only the attachment list, not the entries. The entries are available with object navigation so this shouldn't be hard to work arround in the thunderbird appmodule.

That assumes there is a way to detect which item has the focus, which there may not be. As Marco noted, Mozilla Corporation doesn't work on Thunderbird any more, nor do either of us use Thunderbird ourselves. That said, if someone can file a bug and provide all relevant details, it should be fairly easy for them to fix. Technical info you can provide: if the list items are XUL elements, they can notify accessibility about focus changes using a DOMMenuItemActive event. If they are HTML elements, they can use aria-activedescendant.

**Comment by jcsteh** [comment]
I submitted [a comment on Mozilla bug 1466550](https://bugzilla.mozilla.org/show_bug.cgi?id=1466550#c48) where the initial work on accessibility of the new attachments list was done. However, a separate bug really needs to be filed with proper steps to repro, etc. I haven't done this because I can't devote time to testing, etc.

**Comment by lukaszgo1** [comment]
With latest Thunderbird daily it seems to be partially fixed i.e. after activating the list it is possible to move between attachments with arrow keys and have them read. The list is still announced as a button. I am not a Thunderbird user so it would be nice to have this confirmed by someone else before closing.

**Comment by Adriani90** [comment]
I am testing this in Thunderbird 68.3.0 with NVDA alpha-19420,a2c6f0f8 and cannot reproduce this.
The list is read as a list and the items are reported just fine when navigating with up and down arrow. Also the file name, file size and file source is reported for each item.
It seems the issue is fixed.
Closing as works for me.
Please, if you still have this issue, don'thesitate to comment on this and we can reopen it.

**Comment by lstar4ever** [comment]
Yes, this can be closed, the problem is fixed in the latest Thunderbird version.
